packages/gatsby/src/joi-schemas/joi.js

```diff
@@ -1,10 +1,17 @@
 const Joi = require(`joi`)
 
+const normalizeUrl = url => url.replace(/\/*$/, ``)
```

Comment:
Perhaps consider my method [here](https://github.com/gatsbyjs/gatsby/pull/12128/files#diff-bd517ec758aa05f3e6e1c34b15f65394R3)?

```suggestion
const stripTrailingSlash = chain => chain.replace(/(\w)\/+$/, `$1`)
```

I have a set of unit tests that validate some edge cases, as well (not necessary for this PR).